Update prettier entrypoint.sh to consider dry run

Anu48 opened the pull request:
When dry-run is set to true and we use the `--write` flag, we expect to see what files we'll need to reformat without pushing to remote. However, this requires us to commit the updated changes caused by prettier. Previously, the placement of the dry run logic was done before committing the formatted files, resulting to us not seeing the expected files format. And only using the '--write' flag will not result to a failure even if there are files that requires formatting.

By moving the logic after committing the prettier files, it will show us the comparison between two files, without pushing to remote (given that we use dry flag to true). The dry run will exit with code 1 if there are changes done to the files.

This is the following result using the `--write` flag and dry set to true; when encountering files that need formatting:
https://github.com/GarnerCorp/gala-testing/actions/runs/8802209844 

Versus when we don't have any files to format:
https://github.com/GarnerCorp/gala-testing/actions/runs/8802264992

danielperev commented on scala/action.yml:
```diff
@@ -100,7 +106,7 @@ runs:
     - name: Run build
       shell: bash
-      run: $GITHUB_ACTION_PATH/build-scala-ci.sh ${{ github.event.pull_request.number || github.ref_name }} build
+      run: ${{ inputs.build-script || $GITHUB_ACTION_PATH/build-scala-ci.sh }} ${{ inputs.build-script-args }}
```
1. if `build-script` is set then use `build-script-args` (because if you provide your own script you should provide your own arguments)
2. if `build-script` is not set then try to use `build-script-args` with `$GITHUB_ACTION_PATH/build-scala-ci.sh` 
3. otherwise use default args `${{ github.event.pull_request.number || github.ref_name }}` with `$GITHUB_ACTION_PATH/build-scala-ci.sh`

danielperev commented on scala/action.yml:
```diff
@@ -106,7 +125,7 @@ runs:
     - name: Run build
       shell: bash
-      run: ${{ inputs.build-script || $GITHUB_ACTION_PATH/build-scala-ci.sh }} ${{ inputs.build-script-args }}
+      run: '"${{ inputs.build-script || $GITHUB_ACTION_PATH/build-scala-ci.sh }}" ${{ inputs.build-script-args }}'
```
belongs in previous commit

Anu48 committed:
Update entrypoint.sh to consider dry run

When dry-run is set to true and we use the --write flag, we expect to
see what files we'll need to reformat without pushing to remote.
However, this requires us to commit the updated changes caused by
prettier. Previously, the placement of the dry run logic was done before
committing the formatted files, resulting to us not seeing the expected
files format. And only using the '--write' flag will not result to a
failure even if there are files that requires formatting.

By moving the logic after committing the prettier files, it will show us
the comparison between two files, without pushing to remote
(given that we use dry flag to true).

lzm0 commented on prettier/entrypoint.sh:
```diff
@@ -258,6 +251,14 @@ else
+    echo "Finishing dry-run. Exiting before committing." | tee -a "$GITHUB_STEP_SUMMARY"
+    exit 1
+  fi
+
```
Since we have two `git push` in this workflow, I think it makes sense to extract the dry-run check into a function and call it immediately before each `git push`.

jsoref commented on prettier/entrypoint.sh:
```diff
@@ -169,6 +169,14 @@ prettier_output_summary() {
+      summarize_changes 'Unpretty Files Changes'
+      echo "Finishing dry-run. Exiting before committing." | tee -a "$GITHUB_STEP_SUMMARY"
+      exit 1
+    fi
```
un-indent by 2 spaces

jsoref reviewed (commented):
https://github.com/GarnerCorp/build-actions/blob/9c0188e15ff78a7bf4ee5b44e728f83db962aab4/prettier/action.yml#L42

Should be changed to clarify that the changes that are made will be included in the github job summary.

Anu48 committed:
Update description for dry mode

jsoref commented on prettier/action.yml:
```diff
@@ -39,7 +39,7 @@ inputs:
     default: "**"
   dry:
-    description: Running the script in dry mode just shows whether there are files that should be prettified or not
+    description: Running the script in dry mode just shows whether there are files that should be prettified or not and changes made by prettier will be displayed in github job summary
```
```suggestion
    description: Run prettier in dry-run mode. Display which files are/aren't prettied and changes made by prettier in the GitHub job summary
```

lzm0 replied:
Shouldn't it be `prettified`?

jsoref commented on prettier/action.yml:
```diff
@@ -39,7 +39,7 @@ inputs:
     default: "**"
   dry:
-    description: Running the script in dry mode just shows whether there are files that should be prettified or not
+    description: Run prettier in dry-run mode. Display which files are/aren't prettied and changes made by prettier in the GitHub job summary
```
```suggestion
    description: Run prettier in dry-run mode. Display which files are/aren't pretty and changes made by prettier (to make them pretty) in the GitHub job summary
```

jsoref reviewed (commented):
@lzm0: how about this?
